include/directx/d3dx12.h

```diff
@@ -4037,6 +4038,277 @@ class CD3DX12_NODE_MASK_SUBOBJECT
+        
+        HRESULT result = m_pDevice->CheckFeatureSupport(D3D12_FEATURE_FEATURE_LEVELS, &dFeatureLevels, sizeof(D3D12_FEATURE_DATA_FEATURE_LEVELS));
+
+        MaxSupportedFeatureLevel = dFeatureLevels.MaxSupportedFeatureLevel;
+        return result;
+    }
+
+    // Option 2: Checks if a specific feature level is supported. Returns S_OK if supported and DXGI_ERROR_UNSUPPORTED otherwise.
+    HRESULT FeatureLevelSupported(D3D_FEATURE_LEVEL FeatureLevelRequested)
+    {
+        D3D_FEATURE_LEVEL aFLevels[1];
+        aFLevels[0] = FeatureLevelRequested;
+        D3D12_FEATURE_DATA_FEATURE_LEVELS dFeatureLevels;
+        dFeatureLevels.NumFeatureLevels = 1;
+        dFeatureLevels.pFeatureLevelsRequested = aFLevels;
+
+        HRESULT result = m_pDevice->CheckFeatureSupport(D3D12_FEATURE_FEATURE_LEVELS, &dFeatureLevels, sizeof(D3D12_FEATURE_DATA_FEATURE_LEVELS));
+
+        return result;
+    }
+
+    // 3: Feature Format Support
+    HRESULT FormatSupport(DXGI_FORMAT Format, D3D12_FORMAT_SUPPORT1& Support1, D3D12_FORMAT_SUPPORT2& Support2)
+    {
+        D3D12_FEATURE_DATA_FORMAT_SUPPORT dFormatSupport;
+        dFormatSupport.Format = Format;
+
+        // It is possible that the function call returns an error
+        HRESULT result = m_pDevice->CheckFeatureSupport(D3D12_FEATURE_FORMAT_SUPPORT, &dFormatSupport, sizeof(D3D12_FEATURE_DATA_FORMAT_SUPPORT));
+
+        Support1 = dFormatSupport.Support1;
+        Support2 = dFormatSupport.Support2; // Two outputs. Probably better just to take in the struct as an argument?
+        
+        return result;
+    }
+
+    // 4: Multisample Quality Levels
+    HRESULT MultisampleQualityLevels(DXGI_FORMAT Format, UINT SampleCount, D3D12_MULTISAMPLE_QUALITY_LEVEL_FLAGS Flags, UINT& NumQualityLevels)
+    {
+        D3D12_FEATURE_DATA_MULTISAMPLE_QUALITY_LEVELS dMultisampleQualityLevels;
+        dMultisampleQualityLevels.Format = Format;
+        dMultisampleQualityLevels.SampleCount = SampleCount;
+        dMultisampleQualityLevels.Flags = Flags;
+
+        HRESULT result = m_pDevice->CheckFeatureSupport(D3D12_FEATURE_MULTISAMPLE_QUALITY_LEVELS, &dMultisampleQualityLevels, sizeof(D3D12_FEATURE_DATA_MULTISAMPLE_QUALITY_LEVELS));
+        
+        NumQualityLevels = dMultisampleQualityLevels.NumQualityLevels;
+        return result;
+    }
+
+    // 5: Format Info
+    HRESULT FormatInfo(DXGI_FORMAT Format, UINT8& PlaneCount)
+    {
+        D3D12_FEATURE_DATA_FORMAT_INFO dFormatInfo;
+        dFormatInfo.Format = Format;
+
+        HRESULT result = m_pDevice->CheckFeatureSupport(D3D12_FEATURE_FORMAT_INFO, &dFormatInfo, sizeof(D3D12_FEATURE_DATA_FORMAT_INFO));
+
+        PlaneCount = dFormatInfo.PlaneCount;
+        return result;
+    }
+
+    // 6: GPU Virtual Address Support
+    // TODO: Check if it's exactly the same field as the one in the D3D12_Options feature
+    // FEATURE_SUPPORT_GET(UINT, m_dGPUVASupport, MaxGPUVirtualAddressBitsPerResource);

```

Comment:
There is an entry with the same name in D3D12_Options feature. The two seem to be pointing to the same data field, but I need to doublecheck to make sure they're exactly the same thing.